tools/ocean_neural_network/canyonb.xml

```diff
@@ -0,0 +1,151 @@
+<tool id="bgc_canyon_b" name="Canyon-B" version="@VERSION@+galaxy@VERSION_SUFFIX@" profile="22.05" license="MIT">
+    <description>estimation of ocean CO2 variables and nutrient concentrations</description>
+    <macros>
+        <token name="@VERSION@">0.9.3</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <requirements>
+        <requirement type="package" version="4.3.2">r-base</requirement>
+        <requirement type="package" version="2.13.0">r-r.utils</requirement>
+        <requirement type="package" version="@VERSION@">r-canyonb</requirement>
+        <requirement type="package" version="0.4.0">r-logger</requirement>
+    </requirements>
+    <required_files>
+        <include type="literal" path="run_tool_canyonb.R"/>
+    </required_files>
+    <command detect_errors="exit_code"><![CDATA[
+        mkdir input/ &&
+        cp '$input_data' 'input/$input_data.element_identifier' &&
+        
+        #if $mapping.type == 'no':        
+            Rscript '$__tool_directory__/run_tool_canyonb.R'  
+            'input/$input_data.element_identifier'
+        #else:        
+            Rscript '$__tool_directory__/run_tool_canyonb.R'  
+            'input/$input_data.element_identifier'
+            '$mapping.mapping_pres'
+            '$mapping.mapping_temp'
+            '$mapping.mapping_psal'
+            '$mapping.mapping_doxy'
+            '$mapping.mapping_epres'
+            '$mapping.mapping_etemp'
+            '$mapping.mapping_epsal'
+            '$mapping.mapping_edoxy'
+        #end if     
+    ]]>
+    </command>
+    <inputs>
+        <param name="input_data" type="data" format="csv,txt" label="Input a csv or txt file"/>
+        <conditional name="mapping">
+            <param name="type" type="select" label="Do you have your own mapping to make ?" help="The mapping is needed for ODV collection extracted from the ODV software.">      
+                <option value="no">No, the mapping is already correct in my data</option>
+                <option value="yes">Yes, I have to make a mapping</option>
+            </param>
+            <when value="no">
+            </when>
+            <when value="yes">
+                <param name="mapping_pres" type="text" label="Column name of the pressure"/>
+                <param name="mapping_temp" type="text" label="Column name of the in-situ temperature"/>
+                <param name="mapping_psal" type="text" label="Column name of the salinity"/>
```

Comment:
those are columns from the `input_data`?

If so, the `input_data` needs to be CSV not txt, isn't it?

Reply:
can this be a column_type parameter?

Reply:
nope nope this is something that the user needs to fill in manually according to what his data looks like. I put csv or txt format but it can also be odv format (that galaxy reads as a txt) (maybe one day i'll need to really add this format in the galaxy datatype).

Thus, galaxy will no necessarily register that as a column. If we have time next week maybe we can check it out